tests/nocover/test_statistical_distribution.py

```diff
@@ -416,3 +416,117 @@ def distorted(x):
 test_integers_are_often_small = define_test(
     integers(), 0.2, lambda x: abs(x) <= 100
 )
+
+
+# This series of tests checks that the one_of() strategy flattens branches
+# correctly.  We assert that the probability of any branch is equal
+# (1/8 = 0.125), regardless of how heavily nested it is in the `one_of()`.
+
+# This first strategy chooses an integer between 1 and 8 (inclusive).
+one_of_nested_strategy = one_of(
+    just(1),
+    one_of(
+        just(2),
+        just(3),
+        one_of(
+            just(4),
+            just(5),
+            one_of(
+                just(6),
+                just(7),
+                just(8)
+            )
+        )
+    )
+)
+
+test_one_of_flattens_branches_1 = define_test(
+    one_of_nested_strategy, 0.1, lambda x: x == 1,
+)
+
+test_one_of_flattens_branches_3 = define_test(
+    one_of_nested_strategy, 0.1, lambda x: x == 3,
+)
+
+test_one_of_flattens_branches_5 = define_test(
+    one_of_nested_strategy, 0.1, lambda x: x == 3,
+)
+
+test_one_of_flattens_branches_8 = define_test(
+    one_of_nested_strategy, 0.1, lambda x: x == 8,
+)
+
+
+# This strategy tests interactions with `map()`.  It generates integers from
+# the set {1, 4, 6, 16, 20, 24, 28, 32}.
+double = lambda x: x * 2
+one_of_nested_strategy_with_map = one_of(
+    just(1),
+    one_of(
+        (just(2) | just(3)).map(double),
+        one_of(
+            (just(4) | just(5)).map(double),
+            one_of(
+                (just(6) | just(7) | just(8)).map(double)
+            )
+        ).map(double)
+    )
+)
+
+test_one_of_flattens_mapped_branches_1 = define_test(
+    one_of_nested_strategy_with_map, 0.1, lambda x: x == 1,
+)
+
+test_one_of_flattens_mapped_branches_6 = define_test(
+    one_of_nested_strategy_with_map, 0.1, lambda x: x == 6,
+)
+
+test_one_of_flattens_mapped_branches_20 = define_test(
+    one_of_nested_strategy_with_map, 0.1, lambda x: x == 20,
+)
+
+test_one_of_flattens_mapped_branches_32 = define_test(
+    one_of_nested_strategy_with_map, 0.1, lambda x: x == 32,
+)
+
+
+def fixed_list(x, size):
+    return lists(just(x), min_size=size, max_size=size)
+
+
+# This strategy tests interactions with `flatmap()`.  It generates lists
```

Comment:
`just(None).flatmap`?

I’m not totally sure how flatmap works, so there might be a more natural application of `one_of` to `flatmap` that I’m missing? But this test definitely fails with the release version of Hypothesis and works when I add my flatmap changes.